src/headers.rs

```diff
@@ -62,3 +69,70 @@ pub fn get_user_agent_device(user_agent_client: &Client) -> String {
 
     format!("{device_type}, OS: {device_os}")
 }
+
+#[must_use]
+pub fn get_device_login_event(
+    user_id: i64,
+    ip_address: String,
+    event_type: String,
+    user_agent_client: Option<Client>,
+) -> Option<DeviceLoginEvent> {
+    user_agent_client
+        .map(|client| get_user_agent_device_login_data(user_id, ip_address, event_type, &client))
+}
+
+pub fn get_user_agent_device_login_data(
+    user_id: i64,
+    ip_address: String,
+    event_type: String,
+    user_agent_client: &Client,
+) -> DeviceLoginEvent {
+    let mut model = None;
```

Comment:
```suggestion
    let model = &user_agent_client.device.model.map(|model| model.to_string());
```
A nawet zmienić zamknięcie na funkcję (clippy powinien podpowiedzieć jaką).